### ISSUE TITLE
Win + E still opens a new window if Task Manager is in focus

### ISSUE THREAD

**Issue by i2** [tariibaba/WinENFET#2]
To reproduce:
Open Task Manager and press on Win+E. Even if there is a File Explorer window already open, it will create a new window instead of a new tab.

Update: This is not just limited to Task Manager. Same thing happens if applications like Services, Device Manager, etc. are in focus.

**Comment by tariibaba** [comment]
Hello, thanks for taking the time to contribute!

Try running the application as an administrator.

WinENFET will request administrator rights by default in future versions.

**Comment by i2** [comment]
I wonder if there is another workaround? Some users might not want to run as administrator. Or say if they are using portable AutoHotKey on a computer without admin privileges.

Suggestion regarding future versions: as much as possible, I'd say leave permissions as is.

**Comment by tariibaba** [comment]
Unfortunately, there doesn't seem to be any other way. For some reason, AutoHotKey is simply unable to work properly when any of these apps are in focus.

**Comment by elpocoburrito** [comment]
This issue is caused by windows not letting administrator programs be interacted with using a non-administrator programs. This is a windows security feature as otherwise that could be used to achieve privilege escalation (a virus could obtain administrative access by executing itself through an administrator process that had a legitimate reason to be administrator.
A prime (and simple+relevant) example would an autohotkey script without admin privileges automating the "run" feature of the task manager to run itself, bypassing the UAC prompt.

TL;DR there is no way to fix the issue without running the script as admin. If you found one, I recommend you file that discovery to the Microsoft bug bounty program, I believe they would pay you handsomely (unless you're a security researcher or state sponsored hacker, y'aint finding any ways around it)

**Comment by elpocoburrito** [comment]
So yea, I'd make the program ask for admin permissions automatically and if the user denies it, run the script as a normal user.
I would also offer to memorise the anwser of the user, if they refuse. 
2 simple ways to do so would be to save a true/false in a .ini config file, or to the registry. AHK has built-in support for both so that would be trivial.
For the location of such a config file (.ini route), the working dir would be the easiest, otherwise we could create a folder for the app in %APPDATA%..
I might make a PR for that in the coming week, if you want @tariibaba

**Comment by i2** [comment]
I'd vote for having it in the same folder as opposed to %APPDATA% (for portability and not cluttering)

**Comment by tariibaba** [comment]
Hi @rockbyo5, thank you for you suggestion and contributing to this project.
@i2 Why do you all think it would be a great idea to memorize the users' answers when they deny admin privileges? Considering:
- most programs & installers that request admin privileges don't do this
- admin privileges are simply essential for the program to work correctly and satisfy its requirements ("...**any time** you press Win + E...")

Would love to hear your thoughts on this!

**Comment by elpocoburrito** [comment]
It isn't a bad idea to ask for it everytime and run along if they refuse, i guess, but that could become annoying if a user wants to run it as a normal user (ex: if they dont have admin permission) and also want it to open at boot (which can be achieved as a normal user).
In this use case, they will get a popup for admin permissions at every boot, which can be annoying.

In the vein of having a config file, other configs could be usefull for end users, like for example optionally not having a tray icon when the program is running (with the #notrayicon tag in the script). Having a config file becomes more convenient if the plan is to add multiple config options.

But a simpler option would be for those user to simply alter the (very simple) script source, since it would be a matter of commenting out a single line (or deleting it). That would be nearly as simple as modifying an ini file. We could  make it even simpler by using comments or a very simple "config section" near the top of the script with any desired config having a boolean variable at the top with comments. The only drawback of this route is people need to install autohotkey separately.

A third very simple solution, which would be the most user friendly but which scales poorly if more configs are desired would be to build 2 versions of the script, one for running as admin, one for running as user. That is a fairly common solution I have seen implemented in projects over the years, with a program having 2 exes, one for admin and one for normal use.

Also, the idea of storing the admin decision is actually done in some medium/big programs, TreeSize is one that comes to my mind which prompts for it at every boot. I'm not at home so I cannot grab a screenshot, sadly.
